lib/checkmove.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Cppcheck - A tool for static C/C++ code analysis
+ * Copyright (C) 2007-2016 Cppcheck team.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//---------------------------------------------------------------------------
+// move..
+//---------------------------------------------------------------------------
+
+#include "checkmove.h"
+
+#include "tokenize.h"
+#include "mathlib.h"
+#include "symboldatabase.h"
+#include "astutils.h"
+
+#include <algorithm>
+#include <sstream>
+#include <list>
+#include <cassert>     // <- assert
+#include <cstdlib>
+#include <stack>
+
+//---------------------------------------------------------------------------
+
+// Register this check class (by creating a static instance of it)
+namespace {
+    CheckMove instance;
+}
+
+//---------------------------------------------------------------------------
+
+/** @brief CheckAccessOfMovedVariable */
+void CheckMove::checkAccess()
+{
+    for (const Token *tok = _tokenizer->tokens(); tok; tok = tok->next()) {
+        if (tok->values.empty())
+            continue;
+        for (std::list<ValueFlow::Value>::const_iterator it=tok->values.begin(); it!=tok->values.end(); ++it) {
```

Comment:
The token has : hasKnownIntValue() , getValue() , getMaxValue() , getValueLE(), getValueGE() , getValueTokenMaxStrLength() , getValueTokenMinStrSize() , getValueTokenDeadPointer() .. add one more method here.